Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,63 @@
-function setAlarm() {}
+let countdownTimer;
+let flashInterval;
+
+function setAlarm() {
+  document.getElementById("stop").addEventListener("click", () => {
+    if (flashInterval) {
+      clearInterval(flashInterval);
+      flashInterval = null;
+      document.body.style.backgroundColor = "";
+    }
+  });
+
+  let alarmTime = Number(document.querySelector("input").value);
+
+  if (!alarmTime || isNaN(alarmTime) || alarmTime <= 0) {
+    alert("Please enter a valid positive number for the alarm time");
+    return;
+  }
+
+  let timeLeft = alarmTime;
+
+  function formatTime(time) {
+    let minutes = Math.floor(time / 60);
+    let seconds = time % 60;
+    return `${minutes.toString().padStart(2, "0")}:${seconds
```

Comment:
This is quite a lot of complex code to be inside a string interpolation - it's quite hard to read at a glance what it's doing. Can you think how to make this easier to read?

Also, you have some repeated code here - the 0-padding is the same for both minutes and hours - can you think how to reduce that duplication, e.g. so that if I wanted to change the number of leading zeroes I could do that with just one change not two?